### PR TITLE
prepare metal-ccm for migration to newer metallb versions

### PR DESCRIPTION
**Pull request by mwennrich**
depends on https://github.com/metal-stack/gardener-extension-provider-metal/pull/384

### CODE REVIEW

File: metal/cloud.go [hunk 1]
```diff
@@ -104,12 +109,30 @@ func (c *cloud) Initialize(clientBuilder cloudprovider.ControllerClientBuilder,
 	sshPublicKey := os.Getenv(constants.MetalSSHPublicKey)
 	clusterID := os.Getenv(constants.MetalClusterIDEnvVar)
 
-	k8sClient := clientBuilder.ClientOrDie("cloud-controller-manager")
+	k8sClientSet := clientBuilder.ClientOrDie("cloud-controller-manager")
+	k8sRestConfig, err := clientBuilder.Config("cloud-controller-manager")
+	if err != nil {
+		klog.Error(err)
+	}
+	k8sRestConfig.ContentType = "application/json"
```
Comment (Gerrit91) [inline metal/cloud.go:117]: Just of curiosity... why is this line required?

File: metal/cloud.go [hunk 2]
```diff
@@ -104,12 +109,30 @@ func (c *cloud) Initialize(clientBuilder cloudprovider.ControllerClientBuilder,
 	sshPublicKey := os.Getenv(constants.MetalSSHPublicKey)
 	clusterID := os.Getenv(constants.MetalClusterIDEnvVar)
 
-	k8sClient := clientBuilder.ClientOrDie("cloud-controller-manager")
+	k8sClientSet := clientBuilder.ClientOrDie("cloud-controller-manager")
+	k8sRestConfig, err := clientBuilder.Config("cloud-controller-manager")
+	if err != nil {
+		klog.Error(err)
```
Comment (Gerrit91) [inline metal/cloud.go:115]: Probably we should die on any errors here just like with `ClientOrDie` to prevent the controller from starting up with only half-way initialized clients.

File: pkg/controllers/loadbalancer/metallb.go [hunk 3]
```diff
@@ -131,3 +138,165 @@ func (cfg *MetalLBConfig) ToYAML() (string, error) {
...
+					MatchExpressions: []metav1.LabelSelectorRequirement{{
+						Key:      peer.NodeSelectors[0].MatchExpressions[0].Key,
+						Operator: metav1.LabelSelectorOpIn,
+						Values:   peer.NodeSelectors[0].MatchExpressions[0].Values,
+					}},
+				}},
+			},
+		}
+		res, err := controllerutil.CreateOrUpdate(ctx, c, bgpPeer, func() error {
+			return nil
```
Comment (Gerrit91) [inline pkg/controllers/loadbalancer/metallb.go:194]: I think if the mod func is empty then it will not update anything. Probably you need to set the spec in here.

File: metal/cloud.go (hunk 1 above)
Comment (mwennrich) [reply metal/cloud.go:117]: metallb only supports json:
```
object *v1beta2.BGPPeer does not implement the protobuf marshalling interface and cannot be encoded to a protobuf message
```

**Review by Gerrit91 — APPROVED** [review]
